src/ResourceManager/Network/Commands.Network/Cortex/VirtualWan/P2SVpnServerConfiguration/NewAzureRmVirtualWanP2SVpnServerConfigCommand.cs

```diff
@@ -0,0 +1,285 @@
+            HelpMessage = "The name of the parent VirtualWan this P2SVpnServerConfiguration needs to be associated with.")]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = CortexParameterSetNames.ByVirtualWanName + P2SVpnServerConfigurationParameterSets.RadiusServerConfiguration,
+            HelpMessage = "The name of the parent VirtualWan this P2SVpnServerConfiguration needs to be associated with.")]
+        public string ParentResourceName { get; set; }
+
+        [Alias("ParentVirtualWan", "VirtualWan")]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipeline = true,
+            ParameterSetName = CortexParameterSetNames.ByVirtualWanObject + P2SVpnServerConfigurationParameterSets.Default,
+            HelpMessage = "The VirtualWan this P2SVpnServerConfiguration needs to be associated with.")]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipeline = true,
+            ParameterSetName = CortexParameterSetNames.ByVirtualWanObject + P2SVpnServerConfigurationParameterSets.RadiusServerConfiguration,
+            HelpMessage = "The VirtualWan this P2SVpnServerConfiguration needs to be associated with.")]
+        public PSVirtualWan ParentObject { get; set; }
+
+        [Alias("ParentVirtualWanId", "VirtualWanId")]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = CortexParameterSetNames.ByVirtualWanResourceId + P2SVpnServerConfigurationParameterSets.Default,
+            HelpMessage = "The Id of the parent VirtualWan this P2SVpnServerConfiguration needs to be associated with.")]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = CortexParameterSetNames.ByVirtualWanResourceId + P2SVpnServerConfigurationParameterSets.RadiusServerConfiguration,
+            HelpMessage = "The Id of the parent VirtualWan this P2SVpnServerConfiguration needs to be associated with.")]
+        [ResourceIdCompleter("Microsoft.Network/virtualWans")]
+        [ValidateNotNullOrEmpty]
+        public string ParentResourceId { get; set; }
+
+        [Alias("ResourceName", "P2SVpnServerConfigurationName")]
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The resource name.")]
+        [ValidateNotNullOrEmpty]
+        public virtual string Name { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The list of P2S VPN client tunneling protocols")]
+        [ValidateSet(
+            MNM.VpnGatewayTunnelingProtocol.IkeV2,
+            MNM.VpnGatewayTunnelingProtocol.OpenVPN)]
+        [ValidateNotNullOrEmpty]
+        public string[] VpnProtocol { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
```

Comment:
Please remove ValueFromPipelineByPropertyName from everything except for ResourceId.

Reply:
Done.